src/providers/cryptoauthlib/hash.rs

```diff
@@ -33,4 +34,40 @@ impl Provider {
             }
         }
     }
+
+    pub(super) fn psa_hash_compare_internal(
+        &self,
+        op: psa_hash_compare::Operation,
+    ) -> Result<psa_hash_compare::Result> {
+        let op_hash = op.hash.to_vec();
+        let alg_len = op.alg.hash_length();
+        // calculate input hash
+        let op_compute = psa_hash_compute::Operation {
+            alg: op.alg,
+            input: op.input,
+        };
+        match self.psa_hash_compute_internal(op_compute) {
+            Ok(psa_hash_compute::Result { hash }) => {
+                // compare hashes length
+                if op_hash.len() != alg_len || hash.as_slice().len() != alg_len {
+                    let error = ResponseStatus::PsaErrorInvalidArgument;
+                    format_error!("Hash length comparison failed: ", error);
+                    return Err(error);
+                }
+                // compare hashes
+                if op_hash != hash.as_slice() {
+                    let error = ResponseStatus::PsaErrorInvalidSignature;
+                    format_error!("Hash comparison failed: ", error);
+                    return Err(error);
+                }
+                // return result
+                Ok(psa_hash_compare::Result)
+            }
+            _ => {
+                let error = ResponseStatus::PsaErrorGenericError;
```

Comment:
I think you could return the error from `psa_hash_compute_internal` here instead of `PsaErrorGenericError`. Just doing:
```rust
let hash = self.psa_hash_compute_internal(op_compute)?.hash;
```
is probably fine instead of the match line 49

Reply:
I did it a bit different but code duplication is removed and error type is taken from `psa_hash_compute_internal` result.

Reply:
Thanks :)